### DISCUSSION TITLE
How to send more than 8192 characters when creating a function execution?

### DISCUSSION THREAD

**Discussion by xypmhxy** [appwrite/appwrite#3937] [Q&A] [answered]
### 💭 Description

general_argument_invalid, Invalid data: Value must be a valid string and no longer than 8192 chars (400)
How to solve this problem

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/appwrite/blob/HEAD/CODE_OF_CONDUCT.md)

**Comment by stnguyen90** [comment]
@xypmhxy would you please provide more context like your Appwrite version, what SDK you're using, what version of the SDK, and the code that is causing this?

**Comment by xypmhxy** [comment]
> 

I just call the function normally, but my data is the base64 of the picture, so it is very long, and then this problem occurs, can you solve it?

**Comment by stnguyen90** [comment, marked as answer]
@xypmhxy, I see. You were trying to execute a function and passing an image for the data parameter. I would suggest using the storage API to upload the image. Then, you can either automatically run the function on upload or manually execute the function passing in the file id. The function can then download the file to process.

**Comment by xypmhxy** [comment]
> @xypmhxy, I see. You were trying to execute a function and passing an image for the data parameter. I would suggest using the storage API to upload the image. Then, you can either automatically run the function on upload or manually execute the function passing in the file id. The function can then download the file to process.

My images don't need to be stored, just pass in another third-party api. So this problem is unsolvable, right?

**Comment by stnguyen90** [comment]
@xypmhxy, I still recommend doing it as I suggested. You can delete the file from Appwrite after it's processed.
